app/lib/katello/capsule_content.rb

```diff
@@ -61,6 +80,37 @@ def default_capsule?
       @capsule.default_capsule?
     end
 
+    def current_repositories(environment_id = nil, content_view_id = nil)
+      current_repos = self.pulp_server.resources.repository.retrieve_all
+      katello_repo_ids = []
+      puppet_repo_ids = []
+
+      current_repos.each do |repo|
+        found_repo = Katello::Repository.where(:pulp_id => repo[:id]).first
+        if !found_repo
+          found_puppet = Katello::ContentViewPuppetEnvironment.where(:pulp_id => repo[:id]).first
+          puppet_repo_ids << found_puppet.id if found_puppet
+        else
+          katello_repo_ids << found_repo.id if found_repo
+        end
+      end
+
+      katello_repos = Katello::Repository.where(:id => katello_repo_ids)
+      puppet_repos = Katello::ContentViewPuppetEnvironment.where(:id => puppet_repo_ids)
+
+      if environment_id
+        katello_repos = katello_repos.where(:environment_id => environment_id)
+        puppet_repos = puppet_repos.where(:environment_id => environment_id)
+      end
+
+      if content_view_id
+        katello_repos = katello_repos.in_content_views([content_view_id])
+        puppet_repos = puppet_repos.where(:environment_id => environment_id)
+      end
+
+      katello_repos + puppet_repos
+    end
```

Comment:
I think keeping it in one will help wrt speed. If it was two different methods, I would have to call pulp twice to get the repository list, then cycle both to find the Repositories and the CVPEs as opposed to just once now. I don't know if this would be a trivial difference, but I would be in favor of the longer method if it does make a difference in speed as the size of the repositories scales.